Documentation/operations/upgrade.rst

```diff
@@ -762,6 +768,144 @@ Rollout Instructions
    This will make Cilium read and write identities only to CRDs.
 #. You can now decommission the KVStore.
 
+.. _change_policy_default_local_cluster:
+
+Preparing for a ``policy-default-local-cluster`` change
+#######################################################
+
+Cilium network policies used to implicitly select endpoints from all the clusters.
+In Cilium 1.18, we are introducing a new option called ``policy-default-local-cluster``
+which will be set by default in Cilium 1.19. This option restricts endpoints
+selection to the local cluster by default. If you are using ClusterMesh and
+network policies this will be a **breaking change** and you **need to take action**
+before upgrading to Cilium 1.19.
+
+This new option can be set in the ConfigMap or via the Helm value ``clustermesh.policyDefaultLocalCluster``
+and you can of course still force ``policy-default-local-cluster`` to ``false`` in Cilium 1.19 to keep
+the existing behavior. Note that we will deprecate and eventually remove this option in a future release
+so we encourage you to actively plan your migration to set ``policy-default-local-cluster`` to ``true``.
+
+Migrating network policies in practice
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The command ``cilium clustermesh inspect-policy-default-local-cluster --all-namespaces`` can help you
+discover all the policies that will change as a result of changing ``policy-default-local-cluster``.
+You can also replace ``--all-namespaces`` with ``-n my-namespace`` if you want to only inspect
+policies from a particular namespace.
+
+Let's take an imaginary situation where we would have one network policy to update:
+
+.. code-block:: shell-session
+
+    $ cilium clustermesh prepare-policy-default-local-cluster --all-namespaces
+
+    ⚠️ CiliumNetworkPolicy 0/1
+            ⚠️ default/allow-from-bar
+
+    ✅ CiliumClusterWideNetworkPolicy 0/0
+
+    ✅ NetworkPolicy 0/0
+
+
+In this situation you have only one CiliumNetworkPolicy which will be affected by a
+``policy-default-local-cluster`` change. Let's take a look at the policy:
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2"
+    kind: CiliumNetworkPolicy
+    metadata:
+      name: allow-from-bar
+      namespace: default
+    spec:
+      description: "Allow ingress traffic from bar"
+      endpointSelector:
+        matchLabels:
+          name: foo
+      ingress:
+      - fromEndpoints:
+        - matchLabels:
+            name: bar
+
+This network policy does not explicitly select a cluster. This means that with ``policy-default-local-cluster``
+sets to ``false`` it will allow traffic coming from bar in any clusters connected in your ClusterMesh.
```

Comment:
I wonder whether _foo_ and _bar_ should be somehow emphasized in in the text, to make it clear that they refer to the same foo / bar from the policy. I'm not sure if we have any precedent for this in existing documentation though, so I suppose it's fine to leave it as is.